plugins/backstage-plugin-coder/docs/guides/sdk-recommendations.md

```diff
@@ -0,0 +1,557 @@
+
+Calling the hook will give you an object with all available API methods. As these methods are all async, **none** of them are suitable for use in render logic. They must be called from within effects or event handlers.
+
+```tsx
+// Illustrative example - these patterns are a very bad idea in production!
+function ExampleComponent() {
+  // The workspace type is exported via the plugin
+  const [workspaces, setWorkspaces] = useState<readonly Workspace[]>([]);
+  const sdk = useCoderSdk();
+
+  // The SDK can be called from any effect
+  useEffect(() => {
+    const getInitialWorkspaces = async () => {
+      const workspacesResponse = await sdk.getWorkspaces({ q: 'owner:me' });
+      const workspaces = workspacesResponse.workspaces;
+      setWorkspaces(workspaces);
+    };
+
+    void getInitialWorkspaces();
+
+    // The SDK maintains a stable memory reference; there is no harm in
+    // including it as part of your dependency arrays. In this case, the
+    // dependency array may as well be empty.
+  }, [sdk]);
+
+  return (
+    <>
+      <button
+        // The SDK can also be called from any event handler
+        onClick={async () => {
+          const newWorkspace = await sdk.createWorkspace(
+            'organizationId',
+            'userId',
+            {
+              // Properties for making new workspace go here
+            },
+          );
+
+          setWorkspaces([...workspaces, newWorkspace]);
+        }}
+      >
+        Create new workspace
+      </button>
+
+      <ul>
+        {workspaces.map(workspace => (
+          <li key={workspace.id}>{workspace.name}</li>
+        ))}
+      </ul>
+    </>
+  );
+}
+```
+
+### The SDK object
+
+The SDK object contains all available API methods. All methods follow the format `<verb>` + `<resource name>`. The SDK has these verbs:
+
+- `get`
+- `post`
+- `put`
+- `patch`
+- `upsert`
+- `delete`
+
+Depending on the Coder resource, there may be different API methods that work with a single resource vs all resources (e.g., `sdk.getWorkspace` vs `sdk.getWorkspaces`).
+
+Note that all of these functions will throw an error if the user is not authenticated.
+
+### Error behavior
+
+All SDK functions will throw in the event of an error. You will need to provide additional error handling to expose errors as values within the UI. (Tanstack Query does this automatically for you.)
+
+### Accessing the SDK through `useCoderQuery`
+
+The nuances of how `useCoderQuery` works are covered later in this guide, but for convenience, the hook can access the SDK directly from its `queryFn` function:
+
+```tsx
+const workspacesQuery = useCoderQuery({
+  queryKey: ['workspaces'],
+
+  // Access the SDK without needing to import or call useCoderSdk
+  queryFn: ({ sdk }) => sdk.getWorkspaces({ q: 'owner:me' }),
+});
+```
+
+## Authentication
```

Comment:
I think we could just be prescriptive about using `CoderProvider` especially since we already made it a prerequisite above, which means we would not need to explain the other types and could omit this section.

Especially once we get oauth, it will not be possible for either us nor the user to provide UI for that, and the fallback UI matches best currently with how oauth will behave, I think (since it also uses a popup), so it might be best not to provide any options that will just have to be removed once oauth is available so transitioning is easy.

To be safest we could not export `useCoderAuth` and change our components to also use the fallback UI (which would no longer be a fallback I guess at that point haha), but out of scope for this PR I think.